vocabularies/Org.OData.Temporal.V1.xml

```diff
@@ -67,214 +68,208 @@
+        <Annotation Term="Core.Description" String="Each OData entity maps each point in application time to an instance of the entity type" />
+        <Annotation Term="Core.LongDescription">
+          <String>To address an entity, a point in application time must be specified as described in [OData-Temporal, section 4.2.1].</String>
+        </Annotation>
+      </ComplexType>
 
-      <Term Name="History" Type="Collection(Edm.EntityType)" Nullable="false">
-        <Annotation Term="Core.Description" String="A collection of time slices representing the history of the annotated entity" />
-        <Annotation Term="Core.LongDescription" String="The time slices have the same entity type as the annotated entity and use the same select-expand projection." />
-      </Term>
+      <ComplexType Name="DataHistoryVisible" BaseType="Temporal.DataHistory">
+        <Annotation Term="Core.Description" String="Each OData entity represents data during a period of application time; it is called a *time slice*" />
+        <Annotation Term="Core.LongDescription">
+          <String>The temporal collection MUST NOT contain two entities with the same object key as defined by their `ObjectKey` properties
+          and with overlapping application-time periods as defined by their `TimeSliceStart` and `TimeSliceEnd` properties.
+          The temporal collection always contains all entities (with consecutive time periods) for a given object key.</String>
+        </Annotation>
+        <Property Name="TimeSliceStart" Type="Edm.PropertyPath" Nullable="false">
+          <Annotation Term="Core.Description" String="Property containing lower boundary of a time slice" />
+        </Property>
+        <Property Name="TimeSliceEnd" Type="Edm.PropertyPath" Nullable="false">
+          <Annotation Term="Core.Description" String="Property containing upper boundary of a time slice" />
+        </Property>
+        <Property Name="ObjectKey" Type="Collection(Edm.PropertyPath)" Nullable="false">
+          <Annotation Term="Core.Description" String="The set of primitive properties that identify a temporal object" />
+          <Annotation Term="Core.LongDescription" String="A temporal object is a set of facts whose changes over application time are tracked by the service. The entities in the annotated collection belong to potentially multiple temporal objects, and each temporal object is uniquely identified by the values of the specified object key properties. Object key properties follow the same rules as entity key properties. If no object key is specified, only a single temporal object belongs to the annotated collection." />
+        </Property>
+        <Property Name="SupportedActions" Type="Collection(Core.QualifiedActionName)" Nullable="false">
+          <Annotation Term="Core.Description" String="List of supported temporal actions" />
+        </Property>
+      </ComplexType>
 
+      <ComplexType Name="TimesliceWithPeriod">

```

Comment:
Alternatively, the period boundaries could be brought into a snapshot entity set as instance annotations `@Temporal.ApplicationTimeSupport/DataHistory/TimeSliceStart` and `TimeSliceEnd` if these were of `Edm.PrimitiveType` rather than `Edm.PropertyPath`.

Reply:
Yes, and it would require support of instance annotations in request payloads, which is rare.